mzLib/FlashLFQ/IsoTracker/XIC.cs

```diff
@@ -174,11 +175,11 @@ public double AlignXICs(XIC referenceXIC, int resolution = 1000)
         /// <exception cref="ArgumentException"></exception>
         public void BuildSmoothedCubicSpline(int smoothDegree)
```

Comment:
Collection allocations here need to be worked on. On RT, we call ToArray then ToList later on. On the smoothed intensity, we get an array back from the method and then ToArray is called again. In another case, smoothed intensity comes back as an array, ToList is called, then ToArray is called to send it into another method